src/engine/Engine.cpp

```diff
@@ -3712,3 +3714,82 @@ void Engine::extractBounds( InputQuery &inputQuery )
         }
     }
 }
+
+void Engine::findMissingInputBoundsFromDisjunctions( InputQuery &inputQuery )
```

Comment:
So, this code doesn't really belong here. This is disjunction-specific code, so it belongs in the Disjunction class. I would add an abstract method to PLConstraints, called something like "return minimal LB" and "return maximal UB". Then, make a pass on the constraints, ask them for these bounds, and store them in the input query. I'm not sure whether that should be done in the Engine or in the Preprocessor.